mailbox/api/src/main/java/org/apache/james/mailbox/model/AttachmentId.java

```diff
@@ -20,20 +20,36 @@
 
 
 import java.nio.charset.StandardCharsets;
+import java.util.Optional;
 import java.util.UUID;
 
 import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.james.mime4j.codec.DecodeMonitor;
+import org.apache.james.mime4j.field.ContentTypeFieldImpl;
+import org.apache.james.mime4j.stream.RawField;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Objects;
 import com.google.common.base.Preconditions;
 import com.google.common.base.Strings;
+import com.google.common.primitives.Bytes;
 
 public class AttachmentId {
 
-    public static AttachmentId forPayload(byte[] payload) {
+    private static final String DEFAULT_MIME_TYPE = "text/html";
+
+    public static AttachmentId forPayloadAndType(byte[] payload, String contentType) {
         Preconditions.checkArgument(payload != null);
-        return new AttachmentId(DigestUtils.sha1Hex(payload));
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(contentType));
+        return new AttachmentId(DigestUtils.sha1Hex(Bytes.concat(asMimeType(contentType).getBytes(StandardCharsets.UTF_8), payload)));
```

Comment:
A detail but maybe we can be more efficient by not concatenetaing contentType and payload (I fear a laaaarge array reallocation) but : 

```
SHA1(
   contentType + 
   SHA1(payload))
```

This way the re-allocated array size due to concatenation is neglectable.